tools/clang/unittests/HLSLExec/LongVectors.h

```diff
@@ -293,108 +494,309 @@ template <typename T> struct LongVectorOpTestConfig {
   std::string OperatorString;
   // To be used for the value of -DFUNC
   std::string IntrinsicString;
-  // Optional, can be used to override shader code.
-  bool IsScalarOp = false;
-  bool IsBinaryOp = true;
+  LongVector::BasicOpType BasicOpType = LongVector::BasicOpType_EnumValueCount;
   float Tolerance = 0.0;
-  LongVectorOpType OpType = LongVectorOpType_UnInitialized;
+  LongVector::ValidationType ValidationType = LongVector::ValidationType::ValidationType_Epsilon;
+  LongVectorOpTestConfigTraits<U> OpTypeTraits;
+  std::wstring InputValueSetName1 = L"DefaultInputValueSet1";
+  std::wstring InputValueSetName2 = L"DefaultInputValueSet2";
+  std::wstring InputArgsArrayName = L""; // No default args array
 };
 
-template <typename T> struct LongVectorTestTraits {
-  std::uniform_int_distribution<T> UD = std::uniform_int_distribution(
-      std::numeric_limits<T>::min(), std::numeric_limits<T>::max());
-};
-
-template <> struct LongVectorTestTraits<HLSLHalf_t> {
-  // Float values for this were taken from Microsoft online documentation for
-  // the DirectX HALF data type. HALF is equivalent to IEEE 754 binary 16
-  // format.
-  std::uniform_int_distribution<DirectX::PackedVector::HALF> UD =
-      std::uniform_int_distribution(
-          DirectX::PackedVector::XMConvertFloatToHalf(float(6.10e-5f)),
-          DirectX::PackedVector::XMConvertFloatToHalf(float(65504.0f)));
-};
-
-template <> struct LongVectorTestTraits<HLSLBool_t> {
-  std::uniform_int_distribution<uint16_t> UD =
-      std::uniform_int_distribution<uint16_t>(0u, 1u);
-};
-
-template <> struct LongVectorTestTraits<float> {
-  //  The ranges for generation. A std::uniform_real_distribution can only
-  //  have a range that is equal to the types largest value. This is due to
-  //  precision issues. So instead we define some large values.
-  std::uniform_real_distribution<float> UD =
-      std::uniform_real_distribution(-1e20f, 1e20f);
-};
+template <typename T> bool DoValuesMatch(T A, T B, [[maybe_unused]] float Tolerance, [[maybe_unused]] LongVector::ValidationType ValidationType) {
+  if (Tolerance == 0.0f)
+    return A == B;
 
-template <> struct LongVectorTestTraits<double> {
-  //  The ranges for generation. A std::uniform_real_distribution can only
-  //  have a range that is equal to the types largest value. This is due to
-  //  precision issues. So instead we define some large values.
-  std::uniform_real_distribution<double> UD =
-      std::uniform_real_distribution(-1e100, 1e100);
-};
+  T Diff = A > B ? A - B : B - A;
+  return Diff > Tolerance;
+}
 
-template <typename T> class DeterministicNumberGenerator {
-  // Mersenne Twister 'random' number generator. Generated numbers are based
-  // on the seed value and are deterministic for any given seed.
-  std::mt19937 Generator;
+inline bool DoValuesMatch(HLSLBool_t A, HLSLBool_t B, [[maybe_unused]] float Tolerance, [[maybe_unused]] LongVector::ValidationType ValidationType) { 
+  return A == B;
```

Comment:
Why doesn't this use the Tolerance and ValidationType args? Is it possible to just remove those 2 args?

Reply:
This is an artifact of the templated functions. Theres a compiler warning enabled as a failure for unused function parameters. If I modify the calling location to only include these arguments based on the type it doesn't really solve much. It's called by DoArraysMatch a few lines below.

If you have a cleaner suggestion I'm open!

Reply:
Not necessarily better, but this is another way I think to silence the warning:

```c++
DoValuesMatch(HLSLBool_t A, HLSLBool_t B, float, LongVector::ValidationType) {
```